packages/api/src/services/context/context-service-registry.ts

```diff
@@ -0,0 +1,84 @@
+import type { ContextService } from './context.service';
+import {Service} from '../../providers/service/service';
+import {ValueChangeCallback} from '../../models/context/value-change-callback';
+import {BeforeChangeValidationPromise} from '../../models/context/before-change-validation-promise';
+
+/**
+ * Describes a global subscription to value changes in any registered ContextService.
+ * This is used to apply the same callback logic to all context-managed properties across services.
+ */
+type GlobalSubscription = {
+  callback: ValueChangeCallback<unknown>;
+  beforeChangeValidationPromise?: BeforeChangeValidationPromise<unknown>;
+  afterChangeCallback?: ValueChangeCallback<unknown>;
+};
+
+/**
+ * ContextServiceRegistry is responsible for tracking all registered ContextService instances in the application.
+ * It supports global subscription to context changes and notifies newly registered services of existing global listeners.
+ */
+export class ContextServiceRegistry implements Service {
+  private readonly services: ContextService<Record<string, unknown>>[] = [];
+  private readonly subscribers: GlobalSubscription[] = [];
+
+  registerContextService(service: ContextService<Record<string, unknown>>): void {
+    this.services.push(service);
```

Comment:
I don't like this. We already keep services in the service provider. Why do we need to keep the here as well?!